### PR TITLE
Last validations for jobs

### PR DESCRIPTION
**Pull request by cristianbgp**

### CODE REVIEW

**Review by valevassallo — APPROVED** [review]

**Review by stg101 — APPROVED** [review]

File: app/mailers/rating_mailer.rb [hunk 1]
```diff
@@ -2,32 +2,57 @@ class RatingMailer < ApplicationMailer
     default from: 'notifications@myblog.com'
   
     def report_most_popular_songs
-      @user = params[:user]
       @songs = Song.joins(:ratings).group('songs.id').order('COUNT(ratings.id) DESC').limit(3)
-      mail(to: @user.email, subject: 'Most Popular Songs')
+      new_data = @songs.reduce(""){ | str, song | "#{str},#{song.id}"}
```
Comment (condef5) [inline app/mailers/rating_mailer.rb:6]: You could explain me please what does this line do?

File: app/mailers/rating_mailer.rb [hunk 2]
```diff
@@ -2,32 +2,57 @@ class RatingMailer < ApplicationMailer
...
+      unless $most_popular_songs && $most_popular_songs == new_data
```
Comment (condef5) [inline app/mailers/rating_mailer.rb:7]: Why did you choose to use global variables, what happens if the server restarts?

File: app/mailers/rating_mailer.rb (hunk 1 above)
Comment (cristianbgp) [reply app/mailers/rating_mailer.rb:6]: It stores all the id's of `@songs` in the variable `new_data` as a string

File: app/mailers/rating_mailer.rb (hunk 2 above)
Comment (cristianbgp) [reply app/mailers/rating_mailer.rb:7]: Servers must always be running :joy: 
Actually, it was the easiest way to validate the resource. But we can improve it saving the data in another file.

Comment (condef5) [reply app/mailers/rating_mailer.rb:7]: `Servers must always be running` it's a classic.

**Pull request merged**